### PR TITLE
misc updates

### PR DESCRIPTION
**Pull request by kemalfidan**
This PR includes a lot of small-ish updates to the repo. These are changes I've been looking to make so I bundled them up into 1 PR. 

- Arrays as X[] instead of Array\<X>
- Remove unused tests
- Changed throws from Error to APIError
    - Tests comparing at the Error string work the same
- Mongoose types reference each other 
    - This is for when we want all the vols for an event, etc
- Commented out filledForm field in volunteer
- Updated yarn secrets to pull from kkb email
- Renamed XAttendees to XVolunteers in the Event type

### CODE REVIEW

**Review by rossketron — APPROVED** [review]
Looks like good changes!

File: src/pages/api/volunteers/index.ts [hunk 1]
```diff
@@ -31,10 +31,18 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
             });
         }
     } catch (error) {
-        console.error(error instanceof Error && error);
-        res.status(400).json({
-            success: false,
-            message: (error instanceof Error && error.message) || errors.GENERIC_ERROR,
-        });
+        if (error instanceof APIError) {
```
Comment (zaviermiller) [inline src/pages/api/volunteers/index.ts:34]: seems we use this to check every error, would this be a good in utils.ts?

File: tst/server/Volunteer.test.ts [hunk 2]
```diff
@@ -1,24 +1,16 @@
...
-    return expect(getAlteredUsername("Kemal Fidan")).resolves.toBe("Kemal Fidan123");
-});
-
-test("invalid username", () => {
-    expect.assertions(1);
-    return expect(getAlteredUsername("")).rejects.toThrowError("Invalid username");
-});
-
 // Begin getVolunteer test cases
 describe("getVolunteer() tests", () => {
```
Comment (zaviermiller) [inline tst/server/Volunteer.test.ts:8]: :100:

**Review by zaviermiller — APPROVED** [review]
nice tests, code all looks good

**Pull request merged**